### PR TITLE
[chore] Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jedeen**
Clarifying the process around volunteering and working on issues

### CODE REVIEW

**Comment by jedeen** [comment]
@excaliburjs/core-contributors This is ready for review.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -44,34 +43,48 @@ Please look through our [backlog][issues] to see if your improvement has already
...
 - Send a pull request via Github.
   - Format your pull request title as: [#issue_number] Your commit message (where issue_number is the issue you're closing), and fill out the pull request template that automatically populates the editor window. Please format your pull request title according to our [commit message styleguide](#commit-messages).
 
+#### Deprecating Code
+If you've replaced a piece of Excalibur's API, please mark it as `@obsolete` and provide the new preferred method of performing the same task. Don't forget to include which release it will be removed in! Deprecations are typically performed during the next release, so if your changes are made for the 0.1.0 release, they will be removed in 0.2.0.
+
+example: 
+```
+/** @obsolete use [[SomeClass]].someNewFunction instead **/
+@obsolete({message: 'ex.SomeClass.someFunction is deprecated, and will be removed in 0.2.0',
```
Comment (kamranayub) [inline .github/CONTRIBUTING.md]: Is it capital `Obsolete`? It might be helpful to show the import statement.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -44,34 +43,48 @@ Please look through our [backlog][issues] to see if your improvement has already
...
   - `Changed` for changes in existing functionality
-  - `Deprecated` for features that will be removed in an upcoming release
+  - `Deprecated` for features that will be removed in an upcoming release (see also [deprecating code](#deprecating-code))
   - `Fixed` for bug fixes
 - Please follow our [styleguide](#commit-messages) for your commit messages.
 - Send a pull request via Github.
   - Format your pull request title as: [#issue_number] Your commit message (where issue_number is the issue you're closing), and fill out the pull request template that automatically populates the editor window. Please format your pull request title according to our [commit message styleguide](#commit-messages).
 
+#### Deprecating Code
+If you've replaced a piece of Excalibur's API, please mark it as `@obsolete` and provide the new preferred method of performing the same task. Don't forget to include which release it will be removed in! Deprecations are typically performed during the next release, so if your changes are made for the 0.1.0 release, they will be removed in 0.2.0.
```
Comment (kamranayub) [inline .github/CONTRIBUTING.md]: Mention also probably to call your new method from the old code if applicable

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (eonarheim) [reply .github/CONTRIBUTING.md]: @kamranayub It's lowercase atm, https://github.com/excaliburjs/Excalibur/blob/master/src/engine/Util/Decorators.ts#L18

We could open an issue to make it uppercase `@Obsolete` which seems to be the norm in some libraries (NG being the notable example)

**Pull request merged**